pkg/utils/secrets/static_token.go

```diff
@@ -95,23 +95,40 @@ func (s *StaticTokenSecretConfig) GenerateStaticToken() (*StaticToken, error) {
 	}, nil
 }
 
-// GenerateStaticToken computes a random token of length 64.
-func (s *StaticTokenSecretConfig) AppendStaticToken(staticToken *StaticToken) (*StaticToken, error) {
+// AppendTokens appends required tokens to the existing static tokens list.
+func (b *StaticToken) AppendTokens(s *StaticTokenSecretConfig) (*StaticToken, error) {
```

Comment:
As you are not passing a `StaticToken` but a `StaticTokenSecretConfig` I would rename it:

```suggestion
func (b *StaticToken) GenerateAndAppend(s *StaticTokenSecretConfig) (*StaticToken, error) {
```